File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/operation/meta/value/impl/AttFuncValue.java
package ru.citeck.ecos.records3.record.operation.meta.value.impl;

import org.jetbrains.annotations.NotNull;
import ru.citeck.ecos.commons.utils.func.UncheckedFunction;
import ru.citeck.ecos.records3.record.operation.meta.value.AttValue;

public class AttFuncValue implements AttValue {

    private final UncheckedFunction<String, ?> impl;

    public AttFuncValue(UncheckedFunction<String, ?> impl) {
        this.impl = impl;
    }

    @Override
    public Object getAttribute(@NotNull String name) throws Exception {
        return impl.apply(name);
    }
}

File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/op/atts/service/value/impl/AttFuncValue.kt
package ru.citeck.ecos.records3.record.op.atts.service.value.impl

import ru.citeck.ecos.records3.record.op.atts.service.value.AttValue

class AttFuncValue(val impl: (String) -> Any?) : AttValue {

    override fun getAtt(name: String): Any? {
        return impl.invoke(name)
    }
}
